src/net/sourceforge/kolmafia/textui/parsetree/Value.java

```diff
@@ -271,6 +271,20 @@ public int compareToIgnoreCase(final Value o) {
     return this.compareTo(o, true);
   }
 
+  private static final List<Type> COMPARE_BY_LONG =
```

Comment:
you're only checking containment, so Set is better (assuming types are hashable).